sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -947,8 +951,10 @@ class Dataset[T] private[sql](
    * @group untypedrel
    * @since 2.0.0
    */
-  def join(right: Dataset[_]): DataFrame = withPlan {
-    Join(logicalPlan, right.logicalPlan, joinType = Inner, None, JoinHint.NONE)
+  def join(right: Dataset[_]): DataFrame = withOrigin() {
```

Comment:
since this feature is mostly for ansi mode, do we really need to capture the call side for df APIs that do not create expressions?

Reply:
we don't need to, but please note that query context can be useful with non-ANSI related erros too. here is an example with `over()`: https://github.com/apache/spark/pull/42740/files#diff-728befc0ad63fec7c44a606c62f3f63204293d3ca87da2830fde8ae7a291656bR188-R189

Reply:
Is it possible in a central place? like `withPlan`?

Reply:
Some methods' stacktrace can be captured in `withPlan` but others' can't that's why I decided to add `withOrigin` to all top level methods.